kpi/deployment_backends/kc_access/shadow_models.py

```diff
@@ -256,43 +261,38 @@ class Meta(ShadowModel.Meta):
         db_table = 'auth_user_user_permissions'
 
 
-class KobocatUser(ShadowModel):
-
-    username = models.CharField(_("username"), max_length=30)
-    password = models.CharField(_("password"), max_length=128)
-    last_login = models.DateTimeField(_("last login"), blank=True, null=True)
-    is_superuser = models.BooleanField(_('superuser status'), default=False)
-    first_name = models.CharField(_('first name'), max_length=30, blank=True)
-    last_name = models.CharField(_('last name'), max_length=150, blank=True)
-    email = models.EmailField(_('email address'), blank=True)
-    is_staff = models.BooleanField(_('staff status'), default=False)
-    is_active = models.BooleanField(_('active'), default=True)
-    date_joined = models.DateTimeField(_('date joined'), default=timezone.now)
-
+class KobocatUserProfile(ShadowModel):
+    """
+    From onadata/apps/main/models/user_profile.py
+    Not read-only because we need write access to `require_auth`
+    """
     class Meta(ShadowModel.Meta):
-        db_table = "auth_user"
-
-    @classmethod
-    def sync(cls, auth_user):
-        # NB: `KobocatUserObjectPermission` (and probably other things) depend
-        # upon PKs being synchronized between KPI and KoBoCAT
-        try:
-            kc_auth_user = cls.objects.get(pk=auth_user.pk)
-            assert kc_auth_user.username == auth_user.username
-        except KobocatUser.DoesNotExist:
-            kc_auth_user = cls(pk=auth_user.pk, username=auth_user.username)
+        db_table = 'main_userprofile'
+        verbose_name = 'user profile'
+        verbose_name_plural = 'user profiles'
 
-        kc_auth_user.password = auth_user.password
-        kc_auth_user.last_login = auth_user.last_login
-        kc_auth_user.is_superuser = auth_user.is_superuser
-        kc_auth_user.first_name = auth_user.first_name
-        kc_auth_user.last_name = auth_user.last_name
-        kc_auth_user.email = auth_user.email
-        kc_auth_user.is_staff = auth_user.is_staff
-        kc_auth_user.is_active = auth_user.is_active
-        kc_auth_user.date_joined = auth_user.date_joined
+    # This field is required.
+    user = models.OneToOneField(KobocatUser, related_name='profile')
```

Comment:
@jnm, FYI, I reorganized the class order just because of this relation. 
It was `user = models.OneToOneField(User, related_name='profile')`. 
I needed `KobocatUser` to be declared before `KobocatUserProfile`.